blobs.go

```diff
@@ -66,6 +66,10 @@ type Descriptor struct {
 	// Size in bytes of content.
 	Size int64 `json:"size,omitempty"`
 
+	// ModTime modification time for the file. For backends that
+	// don't have a modification time this may represent the creation time
+	ModTime time.Time
```

Comment:
We should try to stick with the OCI spec for descriptors https://github.com/opencontainers/image-spec/blob/main/descriptor.md, so if we do include this, we should put this in the annotations instead of a new top-level field.

Reply:
I originally was thinking of putting it as an annotation and am totally okay with that. Do you have a recommendation for what key seems appropriate or where the key would be defined?

Reply:
I think the key is alright to just be called `modTime`. 

Not really sure where best to define it, it mostly depends on how this will ultimately get exposed to the user.